lib/OptimizationOptimJL/src/OptimizationOptimJL.jl

```diff
@@ -160,7 +160,7 @@ function SciMLBase.__solve(cache::OptimizationCache{
         if G !== nothing
             cache.f.grad(G, θ, cur...)
             if cache.sense === Optimization.MaxSense
-                G .*= false
+                G .*= -one(eltype(G))
```

Comment:
How does this fix it? It seems to be the same thing in practice. Is there a test?

Reply:
Nope it is not 

```
julia> a = rand(3,3)
3×3 Matrix{Float64}:
 0.929236  0.398606   0.682325
 0.574959  0.978631   0.899744
 0.531948  0.0922131  0.376014

julia> a .* false
3×3 Matrix{Float64}:
 0.0  0.0  0.0
 0.0  0.0  0.0
 0.0  0.0  0.0
```